google/cloud/bigtable/rpc_retry_policy_test.cc

```diff
@@ -50,21 +50,33 @@ void CheckLimitedTime(bigtable::RPCRetryPolicy& tested) {
   //   - We do not care about the results from 40ms to 60ms.
   // I know 10ms feels like a long time, but it is not on a loaded VM running
   // the tests inside some container.
+  // There is one case observed in this which will cause the result fail.
+  //   - Due to server overload if there is a delay between "OnFailure" and
+  //       "now" statement execution then test case may fail. Due to this
+  //       reason we are checking the timestamp before OnFailure and with
+  //       now and both should satify the criteria of the value of "actual"
   auto must_be_true_before =
       start + kLimitedTimeTestPeriod - kLimitedTimeTolerance;
   auto must_be_false_after =
       start + kLimitedTimeTestPeriod + kLimitedTimeTolerance;
+  int true_counter = 0;
+  int false_counter = 0;
   for (int i = 0; i != 100; ++i) {
-    auto actual = tested.on_failure(
+    auto new_start = std::chrono::system_clock::now();
+    auto actual = tested.OnFailure(
```

Comment:
Too much cut&paste?  This class uses  `on_failure` from the days we had not defined a naming convention.